Wikipedia/Code/WMFDatabaseHouseKeeper.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+
+@objc class WMFDatabaseHouseKeeper : NSObject {
+    
+    @objc func performHouseKeepingOnManagedObjectContext(_ moc: NSManagedObjectContext)  throws {
+        
+        try deleteStaleUnreferencedArticles(moc)
+    }
+    
+    private func deleteStaleUnreferencedArticles(_ moc: NSManagedObjectContext) throws {
+        
+        /**
+ 
+        Find `WMFContentGroup`s more than 30 days old.
+ 
+        */
+        
+        let midnightTodayUTC = (Date() as NSDate).wmf_midnightUTCDateFromLocal!
```

Comment:
I'd rather see this as a `guard let midnightTodayUTC = ...` so that it won't crash if for whatever reason this fails. Would be a good idea to add an assertion inside the guard block so that it still would crash on debug builds